src/components/Controls.tsx

```diff
@@ -77,6 +86,19 @@ export default function Controls({ state, setState, mapbox }: ControlsProps) {
     return () => clearTimeout(timer);
   }, [state.input.search]);
 
+  // update url params when search changes
+  useEffect(() => {
+    const { value } = searchInput.current || { value: '' };
+
+    if (searchParams.get('search') !== value && value) {
+      searchParams.set('search', value);
+      setSearchParams(searchParams);
+    } else {
+      searchParams.delete('search');
+      setSearchParams(searchParams);
+    }
+  }, [searchInput.current?.value]);
+
```

Comment:
i think this could be refactored down to 

```
  useEffect(() => {
    if (!searchInput.current) return;
    const { value } = searchInput.current;
    if (searchParams.get('search') === value) return;
    if (value) {
      searchParams.set('search', value);
    } else {
      searchParams.delete('search');
    }
    setSearchParams(searchParams);
  }, [searchInput.current?.value]);
```

Reply:
Nice